### PR TITLE
RETE-22: BE. Posts filter endpoint

### PR DESCRIPTION
**Pull request by Wyla79**

### CODE REVIEW

File: src/main/java/com/laborete/LaboreteAPI/posts/controller/PostsController.java [hunk 1]
```diff
@@ -47,5 +47,10 @@ private ResponseEntity<PostDTO> getPostById(@PathVariable UUID id) {
         return new ResponseEntity<>(this.postsService.getPostById(id), HttpStatus.OK);
     }
 
+    @ApiOperation("Get posts filtered by searchValue")
+    @PostMapping({"/filter"})
+    private List<PostDTO> findByTextLike(@RequestBody String searchValue) {
```
Comment (BobMakhlin) [inline src/main/java/com/laborete/LaboreteAPI/posts/controller/PostsController.java]: In the future the endpoint would provide more functionality (such as ordering posts by fields). So, the method should take a special dto as an argument. Please create a `FilterDto`. At the moment this dto is gonna contain `searchValue` field only, but in the future we would add some more extra fields (like `order` field that specifies fields to order the posts by). Then, we can also rename the method itself - `filterPosts` sounds good for me.

File: src/main/java/com/laborete/LaboreteAPI/posts/repository/PostRepository.java [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
 import com.laborete.LaboreteAPI.posts.entity.PostEntity;
+
+import java.util.List;
 import java.util.UUID;
+
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface PostRepository extends JpaRepository<PostEntity, UUID> {
+
+    List<PostEntity> findByTextContainingIgnoreCase(String text);
```
Comment (BobMakhlin) [inline src/main/java/com/laborete/LaboreteAPI/posts/repository/PostRepository.java:12]: Really good!

File: src/main/java/com/laborete/LaboreteAPI/posts/service/PostsService.java [hunk 3]
```diff
@@ -11,5 +11,8 @@ public interface PostsService {
     PostDTO createPost(CreatePostDTO post);
 
     PostDTO getPostById(UUID id);
+
+    List<PostDTO> getFilteredPosts(String text);
```
Comment (BobMakhlin) [inline src/main/java/com/laborete/LaboreteAPI/posts/service/PostsService.java]: I would simplify the name of the method - `filterPosts`. The current name is good. There is a convension to name a method starting with a verb. Your name follows this rule - _get_. But `filterPosts` sounds a bit more clear to me - we emphasize that the method does filtering of posts inside.

And also, the method should take an instance of type `FilterDto` as a param now.

File: src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java [hunk 4]
```diff
@@ -26,6 +26,7 @@ public class PostsServiceImpl implements PostsService {
     private static final String POST_TEXT_IS_REQUIRED = "Post cannot be without text";
     private static final String USER_CREATOR_IS_REQUIRED = "Post cannot be posted without User";
     private static final String UUID_REQUIRED = "Can`t find post without ID";
+    private static final String SEARCH_FIELD_EMPTY = "Search field is empty";
```
Comment (BobMakhlin) [inline src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java]: Let's rename the constant into `SEARCH_FIELD_IS_EMPTY`.

File: src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java [hunk 5]
```diff
@@ -70,5 +71,10 @@ public PostDTO getPostById(UUID id) {
         return postMapper.postEntityToPostDTO(postEntity);
 
     }
+
+    public List<PostDTO> getFilteredPosts(String text) {
+        List<PostEntity> postEntities = postRepository.findByTextContainingIgnoreCase(text.trim());
```
Comment (BobMakhlin) [inline src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java]: SonarQube recommends to use `var` as much as possible (there are some rare cases when var cannot be used and we have to specify a type of the variable).

**Review by BobMakhlin — APPROVED** [review]

File: src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java (hunk 5 above)
Comment (Wyla79) [reply src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java]: I read about  `var` in Java. Do I understand this remark correctly, if the modified code will look like this:
Записати

Comment (BobMakhlin) [reply src/main/java/com/laborete/LaboreteAPI/posts/service/PostsServiceImpl.java]: Use var whenever it's possible.

**Pull request merged**